### PR TITLE
setup-disk: add 'sys' installation support for s390x

### PR DESCRIPTION
**Pull request by tmh1999**

### CODE REVIEW

**Comment by tmh1999** [comment]
@jirutka Thanks

**Comment by tmh1999** [comment]
It seems like LVM support is not  so complicated. It will push update with LVM support in a couple of  hours.

**Comment by tmh1999** [comment]
https://github.com/alpinelinux/aports/pull/4061

File: setup-disk.in [hunk 1]
```diff
@@ -459,6 +462,7 @@ install_mounted_root() {
 	case "$BOOTLOADER" in
 		grub) setup_grub "$mnt" "$root" "$modules" "$kernel_opts" "$bootdev" $disks ;;
 		syslinux) setup_syslinux "$mnt" "$root" "$modules" "$kernel_opts" "$bootdev" ;;
+		zipl) break;;
```
Comment (jirutka) [inline setup-disk.in]: What loop are you trying to break here? o.O

File: setup-disk.in [hunk 2]
```diff
@@ -994,6 +1006,21 @@ native_disk_install() {
...
 
+		elif [ "$ARCH" = "s390x" ]; then
+			local blocks_per_track=12 tracks_per_cylinder=15
+			local boot_track=$((boot_size * 1024 / 4 / blocks_per_track))
+			local swap_track=$((swap_size * 1024 / 4 / blocks_per_track + boot_track + 1))
+			local conf="$(mktemp)"
+			# DASD devices do not support partition id
+			# assumptions were made that in 'sys' installation, partitions are in order: boot, swap, root
+			# TODO: mdev -s
+			cat > "$conf" <<- EOF
```
Comment (jirutka) [inline setup-disk.in]: Please don’t put space between `<<-` and `EOF`.

Comment (tmh1999) [reply setup-disk.in]: Interesting. I have seen the differences between having space and not but can't remember it. Anyway it seems to work fine and I actually copied this from other place in the same file.

File: setup-disk.in (hunk 1 above)
Comment (tmh1999) [reply setup-disk.in]: Break from the case-esac :D Anyway that was updated in new commit.

**Comment by tmh1999** [comment]
@ncopa we might need this in v3.8.

File: setup-disk.in [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 PREFIX=
 . "$PREFIX/lib/libalpine.sh"
+. "$PREFIX/sbin/setup-dasd"
```
Comment (ncopa) [inline setup-disk.in]: I don't think we should implement this as a `/sbin/setup-dasd`. Maybe better to have it as `/lib/dasd-functions.sh` or similar.

File: setup-disk.in [hunk 4]
```diff
@@ -1266,6 +1307,7 @@ if is_efi || [ -n "$USE_EFI" ]; then
 fi
 
 [ "$ARCH" = "ppc64le" ] && BOOTLOADER=grub
+[ "$ARCH" = "s390x" ] && BOOTLOADER=zipl
```
Comment (ncopa) [inline setup-disk.in]: I'd like:
````sh
case "$ARCH" in
ppc64le) BOOTLOADER=grub;;
s390x) BOOTLOADER=zipl;;
esac
````

**Review by ncopa — CHANGES_REQUESTED** [review]

File: setup-disk.in (hunk 3 above)
Comment (tmh1999) [reply setup-disk.in]: Agreed.

File: setup-disk.in (hunk 4 above)
Comment (tmh1999) [reply setup-disk.in]: Agreed.

**Review by ncopa — APPROVED** [review]

**Comment by tmh1999** [comment]
Great! Let's merge then! Thank you.

**Comment by tmh1999** [comment]
Updated the name change to dasd-function.sh.in

**Comment by tmh1999** [comment]
@jirutka Hi, do you have any other comments on this one ?

**Pull request closed without merging**